src/fund_migration.rs

```diff
@@ -19,25 +24,127 @@ pub enum MigrationStatus {
 }
 
 #[allow(dead_code)]
-fn sign_with_ldk_key(seed: [u8; 64], message: &str) -> Result<(String, String)> {
+pub(crate) async fn migrate_funds(
+    seed: &[u8; 64],
+    data_store: Arc<Mutex<DataStore>>,
+    sdk: &BreezServices,
+) -> Result<()> {
+    if matches!(
+        data_store
+            .lock()
+            .unwrap()
+            .retrive_funds_migration_status()?,
+        MigrationStatus::Completed | MigrationStatus::NotNeeded
+    ) {
+        return Ok(());
+    }
+
+    let (private_key, public_key) = derive_ldk_keys(seed)?;
+    let public_key = public_key.serialize().to_hex();
+
+    let balance = fetch_balance(public_key.clone())?;
+    if balance == 0 {
+        data_store
+            .lock()
+            .unwrap()
+            .append_funds_migration_status(MigrationStatus::NotNeeded)?;
+        return Ok(());
+    }
+
+    data_store
+        .lock()
+        .unwrap()
+        .append_funds_migration_status(MigrationStatus::Pending)?;
+
+    let lsp_info = sdk.lsp_info().await.map_to_runtime_error(
+        RuntimeErrorCode::LspServiceUnavailable,
+        "Failed to get LSP info",
+    )?;
+    let lsp_fee = lsp_info
+        .opening_fee_params_list
+        .get_cheapest_opening_fee_params()
+        .map_to_permanent_failure("Failed to get LSP fees")?;
+    let amout_to_request = add_lsp_fees(balance, &lsp_fee) * 1_000;
```

Comment:
We don't need to implement this. The SDK provides `BreezServices::open_channel_fee()`, right?

Reply:
Forget this, I forgot the lsp fees are calculated on the requested value 🤦

Reply:
As discussed. The assumption here is wrong, because this calculation assumes a payment where the _sent_ amount is an exactly defined number, but what we need for the migration is a payment where the _received_ amount is achieved accurately.